contracts/contracts/CBOR.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+/// While parsing CBOR map, unexpected key
+error CBOR_Error_InvalidKey();
+
+/// While parsing CBOR map, length is invalid, or other parse error
+error CBOR_Error_InvalidMap();
+
+/// While parsing CBOR structure, data length was unexpected
+error CBOR_Error_InvalidLength(uint256);
+
+/// Value cannot be parsed as a uint
+error CBOR_Error_InvalidUintPrefix(uint8);
+
+/// Unsigned integer of unknown size
+error CBOR_Error_InvalidUintSize(uint8);
+
+/// CBOR parsed valid is out of expected range
+error CBOR_Error_ValueOutOfRange();
+
+error CBOR_Error_UintTooLong();
+
+error CBOR_Error_BytesTooLong();
```

Comment:
Missing doc.

```suggestion
/// Too many bytes
error CBOR_Error_BytesTooLong();
```

Reply:
This is a "redundant comment" or more specifically, "comment-name redundancy."

The symbol should be renamed instead.

Which one should we choose, CBOR_Error_TooManyBytes or CBOR_Error_BytesTooLong, which name makes most sense within the context that it's thrown. Perhaps @matevz could even suggest a better error name.

Reply:
I would get rid of it and support 0x5a and 0x5b types too in `encodeBytes()` below.

Reply:
We can't get rid of the error, Solidity has 256bit length prefixes for byte arrays.
I chose an arbitrary cutoff point (64kb), and have no made sure all paths are tested.
Beyond 64kb, user has other problems and is probably doing something wrong.

Reply:
> Solidity has 256bit length prefixes for byte arrays.
> I chose an arbitrary cutoff point (64kb), and have no made sure all paths are tested.
> Beyond 64kb, user has other problems and is probably doing something wrong.

^^ This should go into the error description then:

```solidity
/// @notice Byte array is too long
/// @dev Solidity has 256bit length prefixes for byte arrays. We decided for
///      a reasonable cutoff point (64kb), so at most a 2-byte int describing
///      the array length. Beyond 64kb byte array, user will have other problems
///      and is probably doing something wrong.
error CBOR_Error_BytesTooLong();
```